#167190538 Authenticated User Can Comment on an Existing Article

ayodejiAA opened the pull request:
#### What does this PR do?
- Allows registered users to comment on an existing article
- Allows fetching (Authentication is optional) of all comments on an existing article.

#### Description of Task proposed in this pull request?
- Add route to add comment  -  **POST**  ``` /articles/<slug>/comments``` [Protected Route]
- Add route to get all comments for an article -  **GET**  ``` /articles/<slug>/comments```
- Add comment validation
- Document feature on swagger
- Add migration table for comment

#### How should this be manually tested (Quality Assurance)?
- Checkout to this branch ```ft-user-comment-on-article-167190538```.
- Register or login to get an authentication token. 
- User the add comment route to add a comment. Required: ```comment: "Insert comment"```
- Use the get all comment route to fetch all article comment

#### What are the relevant pivotal tracker stories?
- [#167190538](https://www.pivotaltracker.com/story/show/167190538)

#### Any background context you want to add (Operations Impact)?
Comments allow authors to get feedback on their articles. We all love to know what people think about what we put out to the world.


#### What I have learned working on this feature:
- I learnt Sequelize association

#### Screenshots: 
- Add comment
<img width="1105" alt="Screenshot 2019-08-15 at 12 58 29 PM" src="https://user-images.githubusercontent.com/42908597/63093403-e87e5280-bf5c-11e9-9245-da72f3e9bd6b.png">

 - Get all comments
<img width="1121" alt="Screenshot 2019-08-15 at 1 00 49 PM" src="https://user-images.githubusercontent.com/42908597/63093416-f7fd9b80-bf5c-11e9-82db-05f92d0e293b.png">

 - Article not found
<img width="1124" alt="Screenshot 2019-08-15 at 1 01 06 PM" src="https://user-images.githubusercontent.com/42908597/63093424-05b32100-bf5d-11e9-8028-e17a1b76727e.png">

- Article has no comment
<img width="1118" alt="Screenshot 2019-08-15 at 1 01 32 PM" src="https://user-images.githubusercontent.com/42908597/63093430-09df3e80-bf5d-11e9-89a4-329d32ea7a34.png">

- Mock
<img width="752" alt="Screenshot 2019-08-15 at 3 21 17 PM" src="https://user-images.githubusercontent.com/42908597/63101314-b7a81880-bf70-11e9-8cd5-64f56885ddef.png">

ayodejiAA committed:
featture-[]: Authenticated user can comment on article
- Add route to comment on an article
- Add route to get all articles (Authentication is optional)

chialuka commented on server/controllers/Comments.js:
```diff
@@ -0,0 +1,107 @@
+   * @param {Object} res express response object
+   * @returns {JSON} JSON object with details of new follower
+   */
+  static async getAllCommentsForArticle(req, res) {
```
Why aren't you using a `try-catch` in this controller?

chialuka commented on server/routes/comment.js:
```diff
@@ -0,0 +1,19 @@
+  '/:slug/comments',
+  verifyToken,
+  getSessionFromToken,
+  validateCommentBody,
```
Add `checkUserVerification` middleware to this route

ayodejiAA replied:
Added

chialuka commented on test/comments/comment.test.js:
```diff
@@ -0,0 +1,205 @@
+        });
+      expect(res).to.have.status(201);
+      expect(res.body).to.haveOwnProperty('comment');
+      expect(res.body.comment.user.following).to.be.false; // Not following the author
```
Please remove commented code

ayodejiAA replied:
That is not commented code.

ayodejiAA committed:
feature-[167190538]: Authenticated user can comment on article
- Add route to add comment to article
- Add route to get all comments for an article (Authentication is optional)
- Add comment validation
- Add tests

[Delivers #167190538]

ayodejiAA committed:
feature-[167190538]: Authenticated user can comment on articles

- Add route to add comment to article
- Add route to get all comments for an article (Authentication is optional)
- Add comment validation
- Add tests

[Delivers #167190538]

JuwonAbiola commented on server/routes/index.js:
```diff
@@ -15,5 +16,6 @@ route.use('/auth', auth);
 route.use('/profiles', profile, follower);
 route.use('/user', userFollower);
 route.use('/articles', article);
+route.use('/articles', comment);
```
You can add the comment beside the article instead of repeating route.use ('/articles)

dinobi commented on server/controllers/Comments.js:
```diff
@@ -0,0 +1,114 @@
+  }
+
+  /**
+   * @name getAllCommentsForArticle
```
change to `getArticleComments`

ayodejiAA replied:
Done.
